aristomini/evalui/exam.html

```diff
@@ -0,0 +1,108 @@
+                        {{question.stem}}
+                        {% for choice in question.choices %}
+                            ({{choice.label}}) {{choice.text}}
+                        {% endfor %}
+                    </td>
+                    <td class="td-answer" id="td-{{loop.index0}}-answer">{{question.answerKey}}</td>
+                    <td class="td-solver" id="td-{{loop.index0}}-solver"></td>
+                    <td class="td-correct" id="td-{{loop.index0}}-correct"></td>
+                </tr>
+            {% endfor %}
+
+        </table>
+    </div>
+</div>
+
+    <script type="text/javascript">
+        var exam = {{ exam }};
+        var answerUrl = "{{ solver_url }}/answer";
+
+        function bestResponse(multipleChoiceAnswer) {
+            bestConf = Number.NEGATIVE_INFINITY;
+            bestAnswer = undefined;
+
+            multipleChoiceAnswer.choiceConfidences.forEach(function(cc) {
+                confidence = cc.confidence;
+                if (confidence > bestConf) {
+                    bestConf = confidence;
+                    bestAnswer = cc.choice.label;
+                }
+            })
+
+            return bestAnswer;
+        }
+
+        var numScored = 0
+        var numCorrect = 0
+
+        exam.questions.forEach(function(question, i) {
+            var xhr = new XMLHttpRequest();
+            xhr.open('POST', answerUrl);
+            xhr.setRequestHeader('Content-Type', 'application/json');
+            xhr.onload = function() {
+                if (xhr.status == 200) {
+                    // If you want a more impressive visualization than just
+                    // outputting the raw JSON, change this part of the code.
+                    var result = JSON.parse(xhr.responseText);
+                    var bestAnswer = bestResponse(result.multipleChoiceAnswer);
+                    var correctAnswer = question.answerKey;
+                    var correct = bestAnswer === correctAnswer;
+
+                    // Populate solver cell
+                    document.getElementById("td-" + i + "-solver").innerHTML = bestAnswer;
+                    var correctElement = document.getElementById("td-" + i + "-correct");
+                    if (correct) {
+                        correctElement.classList.remove("incorrect");
+                        correctElement.classList.add("correct");
+                        correctElement.innerHTML = "✔";
+                    } else {
+                        correctElement.classList.add("incorrect");
+                        correctElement.classList.remove("correct");
+                        correctElement.innerHTML = "✗";
+                    }
+
+                    numScored += 1
+                    numCorrect += (correct ? 1 : 0)
+                    pctCorrect = (100 * numCorrect / numScored).toFixed(1) + '%';
+
+                    document.getElementById("score").innerHTML = numCorrect + " / " + numScored + " (" + pctCorrect + ")";
```

Comment:
I suspect there's a race condition in here, because `numScored` and `numCorrect` are updated independently. :-)